ctk/common/src/main/kotlin/org/codice/compliance/utils/XMLDecryptor.kt

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compliance.utils
+
+import org.apache.wss4j.common.crypto.JasyptPasswordEncryptor
+import org.apache.wss4j.common.crypto.Merlin
+import org.apache.xml.security.encryption.XMLCipher
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.codice.compliance.allChildren
+import org.codice.compliance.children
+import org.w3c.dom.Document
+import org.w3c.dom.Element
+import org.w3c.dom.Node
+import java.io.File
+import java.security.Key
+import java.security.Security
+import java.util.Properties
+
+class XMLDecryptor {
+    companion object {
+        // One time initialization
+        init {
+            Security.addProvider(BouncyCastleProvider())
+            org.apache.xml.security.Init.init()
+        }
+
+        private const val ENCRYPTION_FILE_NAME = "encryption.properties"
+        private const val BC_PROVIDER = "BC"
+
+        private val serverPrivateKey by lazy {
+
+            val encryptionFile = File(ClassLoader.getSystemResource(ENCRYPTION_FILE_NAME).toURI())
+            checkNotNull(encryptionFile)
+
+            val encryptionProperties = encryptionFile.inputStream().use {
+                Properties().apply {
+                    load(it)
+                }
+            }
+
+            val keystorePassword = encryptionProperties
+                    .getProperty("org.apache.ws.security.crypto.merlin.keystore.password")
```

Comment:
✏️ You could change `org.apache.ws.security.crypto.merlin.keystore.password` etc. into constants. I think they are used in library